packages/flame/lib/src/game/viewport.dart

```diff
@@ -131,43 +131,54 @@ class FixedResolutionViewport extends Viewport {
   @override
   late Vector2 effectiveSize;
 
-  late Vector2 scaledSize;
-  late Vector2 resizeOffset;
+  final Vector2 scaledSize = Vector2.zero();
+  final Vector2 resizeOffset = Vector2.zero();
   late double scale;
 
+  /// The matrix used for scaling and translating the canvas
+  final Matrix4 _transform = Matrix4.identity();
+
   FixedResolutionViewport(this.effectiveSize);
 
   @override
   void resize(Vector2 newCanvasSize) {
     canvasSize = newCanvasSize;
 
-    final scaleVector = canvasSize.clone()..divide(effectiveSize);
-    scale = math.min(scaleVector.x, scaleVector.y);
-
-    scaledSize = effectiveSize.clone()..scale(scale);
-    resizeOffset = (canvasSize - scaledSize) / 2;
+    scale = math.min(
+      canvasSize.x / effectiveSize.x,
+      canvasSize.y / effectiveSize.y,
+    );
+
+    scaledSize
+      ..setFrom(effectiveSize)
+      ..scale(scale);
+    resizeOffset
+      ..setFrom(canvasSize)
+      ..sub(scaledSize)
+      ..scale(0.5);
   }
 
   @override
   void render(Canvas c, void Function(Canvas) renderGame) {
     c.save();
+    _transform.setIdentity();
+    _transform.translate(resizeOffset.x, resizeOffset.y);
+    _transform.scale(scale);
```

Comment:
I left an review and i think it disappeared 😱 
my suggestion was to have setters for the scale and resizeOffset that precompute the matrix instead of doing that every time

Reply:
Aaah, very good idea, how could I not think of that! :sweat_smile:

Reply:
Do we need setters at all, isn't it enough to change it in `resize`? The other values shouldn't be changed manually right?